### PR TITLE
Update Tally with BillingAccountId 

### PR DESCRIPTION
**Pull request by ksynvrit**
https://issues.redhat.com/browse/ENT-4843

Refactor jpa method to remove long name
Add billingAccountId to ReportCriteria
Add new implementation of Tally api for billingAcctId
Update TallySnapshot with new field
Refactor related unit test

Refactor UsageCalculation with new fields (BillingProvdier & BillingAccountId)
Update test for the SnapshotRollerTester

Refactor Rollup strategy with BillingProvider & BillingAccountId
Add new fields for TagMetaData.
Update Test case for Host.

Refactor UsageCalculation with new fields (BillingProvdier & BillingAccountId)
Update test for the SnapshotRollerTester

Add Billing provider & billingAcctId column to host tally table.
Add new parameters to failing Unit test.
Rework billingAcctId & billingProvider fields in various classes.

### CODE REVIEW

**Review by mstead — CHANGES_REQUESTED** [review]
A few things to update, but shaping up!

File: src/main/java/org/candlepin/subscriptions/rhmarketplace/RhMarketplacePayloadMapper.java [hunk 1]
```diff
@@ -163,7 +171,9 @@ protected List<UsageEvent> produceUsageEvents(TallySummary tallySummary) {
           new UsageCalculation.Key(
               productId,
               ServiceLevel.fromString(snapshot.getSla().toString()),
-              Usage.fromString(snapshot.getUsage().toString()));
+              Usage.fromString(snapshot.getUsage().toString()),
+              BillingProvider.fromString(snapshot.getBillingProvider().toString()),
```
[the remark this answers is not in the document]
Comment (ksynvrit) [reply src/main/java/org/candlepin/subscriptions/rhmarketplace/RhMarketplacePayloadMapper.java:175]: I look at the test and it looks for a subscription that @kahowell has already defaulted to "redhat" in the RhMarketplaceSubscriptionIdProvider. I don't believe I need to add a null billingProvider test case here. Let me know, if you see it differently @mstead

[the remark this answers is not in the document]
Comment (mstead) [reply src/main/java/org/candlepin/subscriptions/rhmarketplace/RhMarketplacePayloadMapper.java:175]: Also, the toString() needs null checked if I'm correct.

[the remark this answers is not in the document]
Comment (lindseyburnett) [reply src/main/java/org/candlepin/subscriptions/rhmarketplace/RhMarketplacePayloadMapper.java:175]: We should have a test for when a TallySummary has a null billing provider.

We also need to protect those `.toString()` invocations from null pointering.

File: src/main/resources/liquibase/202204041547-add-billing-account-id-column-to-snapshot-table.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="202204041547-1" author="ksynvrit">
+        <comment>Add column for billing account id to snapshot table</comment>
+        <addColumn tableName="tally_snapshots">
+            <column name="billing_account_id" type="VARCHAR(255)" />
+        </addColumn>
+    </changeSet>
```
Comment (kahowell) [inline src/main/resources/liquibase/202204041547-add-billing-account-id-column-to-snapshot-table.xml:12]: This should have default `_ANY`, and have a non-null constraint. See `src/main/resources/liquibase/202006180903-add-usage-to-host.xml` for example.

File: src/main/resources/liquibase/202205021439-add-billingProvider-and-billingAccountId-columns-to-host-tally-buckets.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        <addColumn tableName="host_tally_buckets">
+            <column name="billing_provider" type="VARCHAR(255)" />
+        </addColumn>
+    </changeSet>
+
+    <changeSet id="202205021439-2" author="ksynvrit">
+        <comment>Add column for billing account id to host tally buckets table</comment>
+        <addColumn tableName="host_tally_buckets">
+            <column name="billing_account_id" type="VARCHAR(255)" />
+        </addColumn>
```
Comment (kahowell) [inline src/main/resources/liquibase/202205021439-add-billingProvider-and-billingAccountId-columns-to-host-tally-buckets.xml:18]: Both columns should have default `_ANY`, and have a non-null constraint. See `src/main/resources/liquibase/202006180903-add-usage-to-host.xml` for example.

**Review by kahowell — CHANGES_REQUESTED** [review]

File: src/main/resources/liquibase/202204041547-add-billing-account-id-column-to-snapshot-table.xml (hunk 2 above)
Comment (ksynvrit) [reply src/main/resources/liquibase/202204041547-add-billing-account-id-column-to-snapshot-table.xml:12]: Resolved

**Review by kahowell — APPROVED** [review]
LGTM

**Pull request merged**